Establish pattern system

gruetter opened the pull request:
Hi everyone. I have prepared a document to collect and discuss ideas for a pattern system for the InnerSource patterns in preparation for the Dec. 1st patterns group meeting. It includes a summary of related work on patterns in the software space and a proposal for organising our patterns.

I'd like to invite you to add your own proposals and to provide feedback for other proposals in the PR.

>> Georg

Commit:
added document to collect proposals for pattern systems

rrrutledge reviewed (approved):
This looks like a good starting point, especially leading up to Friday's call.  Can anyone else approve also?  I think that one other approval feels like sufficient to merge.  It would be nice to make PRs against this file in _master_.

NewMexicoKid committed:
Update pattern-system.md

Added Iba's paper as a reference on building pattern languages; and also proposed another plane for classifying InnerSource patterns.

dicortazar committed:
Add two extra potential ways to characterize patterns

Commit:
prepared for patterns meeting (added list of patterns for sample classification)

rrrutledge committed:
Update pattern-system.md

rrrutledge commented on pattern-system.md:
```diff
@@ -0,0 +1,273 @@
+
+### Georg Grütter
+
+I propose we use a classification of the problem, a pattern tries to solve as
```
I think a agree with this sentiment but would like to propose that "the problem" be defined as the intersection of vectors in any number of the planes that we're listing out below.

rshanmer reviewed (commented):
The classifications that resonated most with me are Georg's (Getting started, middle mgmt, dev buy-in, fostering contributions, and interfacing).  at my first thought these are helpful as I'm starting a program of inner sourcing.  

However -- who's our intended audience for the patterns?  is it someone just getting started or people in the thick of doing things that want extra help with tools or product development or innovation (e.g. Tim's list)?  

... I still like Georg's.

To add to what Tim wrote -- the GOF book is a collection of patterns.  I agree that we want to create a language (it's cooler and ultimately more useful).  But my experience suggests that we're better off building a catalog first, then reflecting on the catalog and seeing what gaps there are in a language that we need to fill in and then address them to make a language.  My $.02 is that we should keep the idea of a language in the back of our minds and just try to cover the terrain with a rough map that we can fill in between.  When do we start filling in?  Once we are pretty confident that we have basic topography covered (or started) in each of our classification areas.

Commit:
- added categorization proposal. Thanks Ofer!
- added categorizations for Tims and Georgs proposal based on discussion in 2017-12-01 patterns group meeting

NewMexicoKid reviewed (commented):
Looks okay to be one of our published documents; went ahead and merged this. We can later edit this and/or augment it with other documents.
